src/lib.rs

```diff
@@ -114,97 +114,3630 @@ use std::{
     ffi::OsString,
     fmt::{self, Display, Formatter},
     io::{Error, ErrorKind},
+    env,                    // for environment variables (LANG, LC_ALL)
 };
 
+/// output : en_US.UTF-8
+/// en     : language
+/// US     : country
+#[allow(dead_code)]
+fn get_language_and_country() -> String {
+    if let Ok(language) = env::var("LC_ALL") {
+        // possible output : en_US.UTF-8
+        return language;
+    } else if let Ok(language) = env::var("LANG") {
+        // possible output : en_US.UTF-8
+        return language;
+    } else if let Ok(language) = env::var("LANGUAGE") {
+        // possible output : en_US.UTF-8
+        return language;
+    }
+    String::new()
+}
```

Comment:
For this change, could you split it out into it's own PR?

This PR is already large, and it's easier to review in smaller chunks.